lib/Doctrine/CouchDB/HTTP/SocketClient.php

```diff
@@ -99,7 +105,7 @@ protected function checkConnection()
      * @param array $headers
      * @return string
      */
-    protected function buildRequest( $method, $path, $data, $headers)
+    public function buildRequest( $method, $path, $data = null, array $headers = array())
```

Comment:
If we're making this method public, that means you're going to use it publicly in the code somewhere. Then we should also add this to the `Client.php` interface and document it there.